x-pack/plugins/security/server/routes/authentication/common.ts

```diff
@@ -135,4 +141,24 @@ export function defineCommonRoutes({ router, authc, basePath, logger }: RouteDef
       }
     })
   );
+
+  router.post(
+    { path: '/internal/security/access_agreement/acknowledge', validate: false },
+    createLicensedRouteHandler(async (context, request, response) => {
+      // If license doesn't allow access agreement we shouldn't handle request.
+      if (!license.getFeatures().allowAccessAgreement) {
+        logger.warn(`Attempted to acknowledge access agreement when license doesn allow it.`);
+        return response.notFound();
```

Comment:
__note:__ I was considering two options here, either to return 403 or 404. I picked 404 and the rational is that license doesn't provide this feature and that basically means that feature is missing (not found).. What do you think?

Reply:
Okay, makes sense too, let's go with `403`. By reason you mean our own custom message, right? Currently `SecurityLicense` doesn't keep the optional message that Licensing plugin would return in case of failed check...

Reply:
Discussed over Slack, we'll go with custom `Current license doesn't support access agreement.` message.